src/protocol/http/src/main/java/org/apache/jmeter/protocol/http/correlation/Correlation.java

```diff
@@ -0,0 +1,560 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.jmeter.protocol.http.correlation;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URLEncoder;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.swing.tree.TreePath;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.NameValuePair;
+import org.apache.http.client.utils.URLEncodedUtils;
+import org.apache.jmeter.config.Argument;
+import org.apache.jmeter.config.Arguments;
+import org.apache.jmeter.exceptions.IllegalUserActionException;
+import org.apache.jmeter.extractor.gui.BoundaryExtractorGui;
+import org.apache.jmeter.extractor.gui.HtmlExtractorGui;
+import org.apache.jmeter.extractor.gui.RegexExtractorGui;
+import org.apache.jmeter.extractor.gui.XPath2ExtractorGui;
+import org.apache.jmeter.extractor.json.jsonpath.gui.JSONPostProcessorGui;
+import org.apache.jmeter.gui.CorrelationTableModel;
+import org.apache.jmeter.gui.GuiPackage;
+import org.apache.jmeter.gui.tree.JMeterTreeNode;
+import org.apache.jmeter.protocol.http.control.Header;
+import org.apache.jmeter.protocol.http.control.HeaderManager;
+import org.apache.jmeter.protocol.http.correlation.extractordata.BoundaryExtractorData;
+import org.apache.jmeter.protocol.http.correlation.extractordata.ExtractorData;
+import org.apache.jmeter.protocol.http.correlation.extractordata.HtmlExtractorData;
+import org.apache.jmeter.protocol.http.correlation.extractordata.JsonPathExtractorData;
+import org.apache.jmeter.protocol.http.correlation.extractordata.RegexExtractorData;
+import org.apache.jmeter.protocol.http.correlation.extractordata.XPath2ExtractorData;
+import org.apache.jmeter.protocol.http.sampler.HTTPSamplerBase;
+import org.apache.jmeter.protocol.http.sampler.HTTPSamplerProxy;
+import org.apache.jmeter.protocol.http.util.HTTPArgument;
+import org.apache.jmeter.protocol.http.util.HTTPConstants;
+import org.apache.jmeter.save.SaveService;
+import org.apache.jmeter.testelement.TestElement;
+import org.apache.jmeter.testelement.property.PropertyIterator;
+import org.apache.jmeter.util.JMeterUtils;
+import org.apache.jmeter.visualizers.CorrelationRecorder;
+import org.apache.jorphan.collections.HashTree;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class Correlation {
```

Comment:
Can you add some javadoc describing class role

Reply:
Updated the java doc.

Reply:
We have updated the javadoc to describe javadoc in detail.